libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -2326,17 +2332,17 @@ genNumList ::
   , MaybeBounded a
   , Foldy fn a
   ) =>
-  Spec fn a ->
-  Spec fn a ->
+  Specification fn a ->
+  Specification fn a ->
   GenT m [a]
 genNumList elemSIn foldSIn = do
   normElemS <- normalize elemSIn
   normFoldS <- normalize foldSIn
   explain
     [ "Can't generate list of ints with fold constraint"
-    , show $ "  elemSpec =" <+> pretty elemSIn
-    , show $ "  normElemSpec =" <+> pretty normElemS
-    , show $ "  foldSpec =" <+> pretty foldSIn
+    , "  elemSpec = " ++ show elemSIn
+    , "  normElemSpec = " ++ show normElemS
+    , "  foldSpec = " ++ show foldSIn
     ]
```

Comment:
Not a good idea in my opinion to change Pretty to Show.

Reply:
This is just a refactoring:
```haskell
instance HasSpec fn a => Show (Specification fn a) where
  showsPrec d = shows . pretty . WithPrec d
```